gwpy/timeseries/timeseries.py

```diff
@@ -1463,18 +1474,26 @@ def taper(self, side='leftright'):
         if side not in ('left', 'right', 'leftright'):
             raise ValueError("side must be one of 'left', 'right', "
                              "or 'leftright'")
+        if duration and nsamples:
+            raise ValueError("only one of 'duration' or 'nsamples' may be "
+                             "provided as arguments")
+        elif duration:
+            nsamples = int(duration * self.sample_rate.to("Hz").value)
         out = self.copy()
+        # if a duration or number of samples is not specified, automatically
         # identify the second stationary point away from each boundary,
         # else default to half the TimeSeries width
         nleft, nright = 0, 0
-        mini, = signal.argrelmin(out.value)
-        maxi, = signal.argrelmax(out.value)
+        if not nsamples:
+            mini, = signal.argrelmin(out.value)
+            maxi, = signal.argrelmax(out.value)
         if 'left' in side:
-            nleft = max(mini[0], maxi[0])
-            nleft = min(nleft, self.size/2)
+            nleft = nsamples if nsamples else max(mini[0], maxi[0])
```

Comment:
@tjma12, this can be simplified somewhat, which may appease Codeclimate:

```suggestion
            nleft = nsamples or max(mini[0], maxi[0])
```